### PR TITLE
Fix multiple mentors added to matcher

### PR DESCRIPTION
**Pull request by ericyche**
Matcher would create multiple mentors with the same user. We turn the input list into a set so there cannot be any duplicate emails. 
https://pylint.pycqa.org/en/latest/user_guide/configuration/all-options.html more details about flags i added. 
Screenshot 2023-09-03 at 11 11 19 PM
Screenshot 2023-09-03 at 11 10 57 PM

### CODE REVIEW

**Comment by ericyche** [comment]
i am never safe from pre-commit

File: csm_web/scheduler/views/matcher.py [hunk 1]
```diff
@@ -520,21 +518,24 @@ def run_matcher(course: Course):
         - unmatched: [int, int, ...] of mentor ids
     """
     # get slot information
-    slots = MatcherSlot.objects.filter(matcher=course.matcher)
+    matcher_slots = MatcherSlot.objects.filter(matcher=course.matcher)
     # get preference information
-    preferences = MatcherPreference.objects.filter(slot__matcher=course.matcher)
```
Comment (smartspot2) [inline csm_web/scheduler/views/matcher.py:525]: This should be `matcher_preferences` (plural) to match the other variables (and the semantics of this variable)

**Review by smartspot2 — CHANGES_REQUESTED** [review]
Fix the one variable name and I'll approve the next revision.

**Review by ericyche — COMMENTED** [review]
idk what this page is on github but i will add a comment.

**Review by smartspot2 — CHANGES_REQUESTED** [review]
Please squash and remove the merge commit.

**Comment by ericyche** [comment]
not fan of merge commit :(

**Review by edwardneo — APPROVED** [review]
Nice!

**Review by smartspot2 — APPROVED** [review]

**Pull request merged**